dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedInterval.java

```diff
@@ -70,6 +72,13 @@ public ZonedDateTime getEndInclusiveUTC() {
         return getEndInclusive().withZoneSameInstant(Z_UTC);
     }
 
+    @Nonnull
+    public ZonedInterval span(@Nonnull ZonedInterval interval) {
+        ZonedDateTime startTime = Collections.min(Arrays.asList(interval.getStart(), this.getStart()));
+        ZonedDateTime endTime = Collections.max(Arrays.asList(interval.getEndExclusive(), this.getEndExclusive()));
```

Comment:
nit: I think it can be done in a more direct way without wrapping it into collection and then getting max from this collection

```
ZonedDateTime startTime = interval.getStart().isBefore(this.getStart()) ? interval.getStart() : this.getStart()
ZonedDateTime endTime = interval.getEndExclusive().isAfter(this.getEndExclusive()) ? interval.getEndExclusive() : this.getEndExclusive()

```
wdyt?

Reply:
There's also `ObjectUtils.min()`

Reply:
Awesome, applied!